python-checks/src/main/java/org/sonar/python/checks/ConstantConditionCheck.java

```diff
@@ -0,0 +1,156 @@
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NONE;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NOT;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.NUMERIC_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.OR;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.STRING_LITERAL;
+import static org.sonar.plugins.python.api.tree.Tree.Kind.UNPACKING_EXPR;
+
+@Rule(key = "S5797")
+public class ConstantConditionCheck extends PythonVisitorCheck {
+
+  private static final String MESSAGE = "Replace this expression; used as a condition it will always be constant.";
+
+  @Override
+  public void visitIfStatement(IfStatement ifStatement) {
+    checkConstantCondition(ifStatement.condition());
+    scan(ifStatement.body());
+    scan(ifStatement.elifBranches());
+    scan(ifStatement.elseBranch());
+  }
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpression conditionalExpression) {
+    checkConstantCondition(conditionalExpression.condition());
+    super.visitConditionalExpression(conditionalExpression);
+  }
+
+  private void checkConstantCondition(Expression condition) {
+    Expression constantBooleanExpression = getConstantBooleanExpression(condition);
+    if (constantBooleanExpression != null) {
+      addIssue(constantBooleanExpression, MESSAGE);
+    }
+    checkExpression(condition);
+  }
+
+  private static boolean isConstant(Expression condition) {
+    return TreeUtils.isBooleanLiteral(condition) ||
+      condition.is(NUMERIC_LITERAL, STRING_LITERAL, NONE) ||
+      isConstantCollectionLiteral(condition);
+  }
+
+  private static Expression getConstantBooleanExpression(Expression condition) {
+    if (condition.is(AND, OR)) {
+      BinaryExpression binaryExpression = (BinaryExpression) condition;
+      if (isConstant(binaryExpression.leftOperand())) {
+        return binaryExpression.leftOperand();
+      }
+      if (isConstant(binaryExpression.rightOperand())) {
+        return binaryExpression.rightOperand();
+      }
+    }
+    if (condition.is(NOT) && isConstant(((UnaryExpression) condition).expression())) {
+      return ((UnaryExpression) condition).expression();
+    }
+    return null;
+  }
+
+  private static boolean isConstantCollectionLiteral(Expression condition) {
+    switch (condition.getKind()) {
+      case LIST_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((ListLiteral) condition).elements());
+      case DICTIONARY_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((DictionaryLiteral) condition).elements());
+      case SET_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((SetLiteral) condition).elements());
+      case TUPLE:
+        return isAlwaysEmptyOrNonEmptyCollection(((Tuple) condition).elements());
+      default:
+        return false;
+    }
+  }
+
+  private static boolean isAlwaysEmptyOrNonEmptyCollection(ExpressionList elements) {
```

Comment:
I think you could remove this method if you call `ExpressionList#expressions` directly on line 98.

Reply:
Indeed, good point! thanks!